File: DSA Using C++/2. Algorithms/Sorting Algorithms/Insertion_sort.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cout << "Enter size of array: ";
    cin >> n;
    int arr[n];
    cout << "Enter elements of array \n";
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    cout << "Given array is \n";
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;

    for (int i = 1; i < n; i++)
    {
        int current = arr[i];
        int j = i - 1;
        while (arr[j] > current && j >= 0)
        {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = current;
    }

    cout << "Sorted array is: \n";
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}

File: DSA Using Java/Algorithms/Sorting Algorithms/InsertionSort.java
public class InsertionSort {
    public static void printArray(int[] arr){
        for (int i : arr){
            System.out.print(i+" ");
        }
        System.out.println();
    }
    public static void main(String[] args) {
        int[] arr = {12, 76, 9, 3, 4, 7, 1, 2, 87, 34, 2, 3};
        for (int i = 0; i < arr.length; i++){
            int current = arr[i];
            int j = i-1;
            while (j >= 0 && current < arr[j]){
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = current;
        }
        printArray(arr);
    }
}
